### ISSUE TITLE
kernel-controllers(memcfg_stress_test)Test causes system to freeze

### ISSUE THREAD

**Issue by 843723683** [linux-test-project/ltp#632]
Hi：
     8G memory on Dell desktop. Tested on rhat7.6 and centos7.6 respectively.
     All stuck in the memcfg_stress_test test item. No mouse or keyboard response

command: ./runltp -f controllers

**Comment by 843723683** [comment]
> Hi：
> 8G memory on Dell desktop. Tested on rhat7.6 and centos7.6 respectively.
> All stuck in the memcfg_stress_test test item. No mouse or keyboard response
> 
> command: ./runltp -f controllers

**Comment by metan-ucw** [comment]
That sounds like the system went out of memory and started trashing. I guess that the problem here is that the calculation for a amount of memory to be used by the test is (free ram + free swap)/2, which may return too much on systems where swap is more or less equal to a system ram in size.

**Comment by pevik** [comment]
Maybe change `testcases/kernel/controllers/memcg/stress/memcg_stress_test.sh`
```
-       MEM=$(( $mem_free + $swap_free / 2 ))
+       MEM=$(( $mem_free + $swap_free / 5 ))
```
@843723683 Could you please test it (https://github.com/pevik/ltp/commit/2540f102c28942bfea90d01508a7e410dce46f3e from my branch on https://github.com/pevik/ltp/tree/gh632).

**Comment by metan-ucw** [comment]
I do not think that dividing the value with greater value would hold a candle. I guess that the available memory should be mostly based on $mem_free, we may add a little more if there is enough swap, but not too much.

**Comment by pevik** [comment]
I'd be for `MEM=$(( $mem_free + $swap_free / 10 ))`. BTW what is your name (for `Reported-by: Your Name <your@email.com>` tag?, see http://elinux.org/Developer_Certificate_Of_Origin)

**Comment by pevik** [comment]
@metan-ucw ok to put this simple fix into upcoming release?

**Comment by 843723683** [comment]
Thanks for adopting this suggestion.
Reported-by: Zuo Liu<lz843723683@gmail.com>

**Comment by pevik** [comment]
Thanks! Posted to ML
https://lists.linux.it/pipermail/ltp/2020-January/015080.html
https://patchwork.ozlabs.org/patch/1224644/

**Comment by wangli5665** [comment]
I just wondering what will be happening if the size of swap_free is close to 0. Zuo Liu, could you help to test this patch with the swap disable?

**Comment by 843723683** [comment]
@wangli5665 Ok, tomorrow i will do a set of swap = 0 experiments

**Comment by 843723683** [comment]
@wangli5665 friend. 

`MEM=$(( $mem_free + $swap_free / 10 ))`
> swapoff -a
    cat /proc/meminfo | grep SwapFree
          SwapFree: 0 kb

The result is the same as MEM = $ (($ mem_free)).
Keyboard and mouse can be used normally

**Comment by pevik** [comment]
BTW patch https://patchwork.ozlabs.org/patch/1224644/ not merged as @metan-ucw suggest a smarter solution:
https://patchwork.ozlabs.org/patch/1224644/#2346786
> I wonder if we can be a bit smarter about this. Maybe we should add some percentage of RAM if enough swap is present.

But I'm not sure what is a goal of this RAM + swap balance. Using as much swap as possible, but still safe (i.e. used swap needs to be still less than free memory)?
